scripts/release.py

```diff
@@ -0,0 +1,244 @@
+    with PYPROJECT_TOML_PATH.open("rb") as pyproject_file:
+        pyproject_toml = tomlkit.load(pyproject_file)
+
+    project_table = pyproject_toml["project"]
+    assert isinstance(project_table, tomlkit.items.Table)
+
+    project_table["version"] = tomlkit.string(str(versions.new_vscode_version))
+
+    existing_dependencies = project_table["dependencies"]
+    assert isinstance(existing_dependencies, tomlkit.items.Array)
+    assert len(existing_dependencies) == 3
+    existing_dependencies[1] = tomlkit.string(f"ruff-lsp=={versions.latest_ruff_lsp}")
+    existing_dependencies[2] = tomlkit.string(f"ruff=={versions.latest_ruff}")
+
+    with PYPROJECT_TOML_PATH.open("w") as pyproject_file:
+        tomlkit.dump(pyproject_toml, pyproject_file)
+
+
+def bump_package_json_version(new_version: Version) -> None:
+    """Update the version of this package in `package.json`."""
+    with PACKAGE_JSON_PATH.open("rb") as package_json_file:
+        package_json = json.load(package_json_file)
+    package_json["version"] = str(new_version)
+    with PACKAGE_JSON_PATH.open("w") as package_json_file:
+        json.dump(package_json, package_json_file, indent=2)
+        package_json_file.write("\n")
+
+
+def update_readme(latest_ruff: Version) -> None:
+    """Ensure the README is up to date with respect to our pinned Ruff version."""
+    readme_text = README_PATH.read_text()
+    readme_text = re.sub(
+        r"The extension ships with `ruff==\d\.\d\.\d`\.",
+        f"The extension ships with `ruff=={latest_ruff}`.",
+        readme_text,
+    )
+    readme_text = re.sub(
+        r"ruff/\d\.\d\.\d\.svg", f"ruff/{latest_ruff}.svg", readme_text
+    )
+    README_PATH.write_text(readme_text)
+
+
+def update_changelog(versions: RuffVersions) -> None:
+    """Add a changelog entry describing the updated dependency pins."""
+    with CHANGELOG_PATH.open() as changelog_file:
+        changelog_lines = list(changelog_file)
+
+    assert (
+        changelog_lines[4] == f"## {versions.existing_vscode_version}\n"
+    ), f"Unexpected content in CHANGELOG.md ({changelog_lines[4]!r}) -- perhaps the release script is out of date?"
+
+    if (
+        versions.latest_ruff != versions.existing_ruff_pin
+        and versions.latest_ruff_lsp != versions.existing_ruff_lsp_pin
+    ):
+        changelog_entry_middle = (
+            f"This release upgrades the bundled Ruff version to `v{versions.latest_ruff}`, and the bundled `ruff-lsp` "
+            f"version to `{versions.latest_ruff_lsp}`."
+        )
+    elif versions.latest_ruff != versions.existing_ruff_pin:
+        changelog_entry_middle = f"This release upgrades the bundled Ruff version to `v{versions.latest_ruff}`."
+    elif versions.latest_ruff_lsp != versions.existing_ruff_lsp_pin:
+        changelog_entry_middle = f"This release upgrades the bundled `ruff-lsp` version to `v{versions.latest_ruff_lsp}`."
+    else:
+        changelog_entry_middle = ""
+
+    changelog_entry = textwrap.dedent(f"""\
```

Comment:
Why do you need the `\`. 

Isn't it nice how Ruff wraps the multiline string? :)

Reply:
It's a very common practice when using `textwrap.dedent()` in combination with triple-quoted strings, or it's easy to end up with a dedented string where the first line has odd indentation https://grep.app/search?q=textwrap.dedent%28%22%22%22%5C&filter[lang][0]=Python

> Isn't it nice how Ruff wraps the multiline string? :)

I'm... actually not a massive fan of that formatting choice 🙈 but I don't care enough to argue about it 😆

Reply:
I know it is... but it's something black doesn't support :)